### PR TITLE
Sanitize metadata key

### PR DESCRIPTION
**Pull request by aerlaut**
# Background
#### Link to issue 

https://this-is-biomage.slack.com/archives/C015CHSUDRU/p1622644162098700

#### Link to staging deployment URL 

in UI PR

#### Links to any Pull Requests related to this

UI : https://github.com/biomage-ltd/ui/pull/302
API : https://github.com/biomage-ltd/api/pull/142

#### Anything else the reviewers should know about the changes here

- Check for `-` in metadata key before passing data into GEM2S

# Changes
### Code changes

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [ ] Unit tests written
- [X] Tested locally with Inframock (with latest production data downloaded with biomage experiment pull)
- [X] Deployed to staging

To set up easy local testing with inframock, follow the instructions here: https://github.com/biomage-ltd/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/biomage-ltd/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [ ] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [ ] After the PR is approved, the `unstage` script in here: https://github.com/biomage-ltd/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [ ] Photo of a cute animal attached to this PR

### CODE REVIEW

File: src/api/general-services/pipeline-manage/index.js [hunk 1]
```diff
@@ -249,7 +249,11 @@ const createGem2SPipeline = async (experimentId) => {
 
   if (metadataKeys.length) {
     taskParams.metadata = metadataKeys.reduce((acc, key) => {
-      acc[key] = samples.ids.map(
+      // Make sure the key does not contain '-' as it will cause failure in GEM2S
+      const sanitizedKey = key.replace(/-+/g, '_');
+
+      acc[sanitizedKey] = samples.ids.map(
+        // Fetch using unsanitized key as it is the key used to store metadata in sample
         (sampleUuid) => samples[sampleUuid].metadata[key] || defaultMetadataValue,
```
Comment (xverges) [inline src/api/general-services/pipeline-manage/index.js:257]: Shouldn't we be returning `acc`? (I hate reduce, BTW :-D )

Comment (aerlaut) [reply src/api/general-services/pipeline-manage/index.js:257]: It's returned on line 259, you have to scroll down a bit to see it.

Comment (aerlaut) [reply src/api/general-services/pipeline-manage/index.js:257]: It takes sometime to get to know `reduce`, but once you get the hang of it, it's pretty powerful (albeit difficult to read)

Comment (xverges) [reply src/api/general-services/pipeline-manage/index.js:257]: Oh, sorry! Thanks!

**Review by xverges — APPROVED** [review]

**Pull request merged**